mir-ci/mir_ci/tests/test_mir_flutter_app.py

```diff
@@ -0,0 +1,55 @@
+import itertools
+from pathlib import Path
+from typing import Collection
+
+import pytest
+from mir_ci import VARIANT
+from mir_ci.fixtures import apps, servers
+from mir_ci.program.app import App
+from mir_ci.program.display_server import DisplayServer
+from mir_ci.wayland.screencopy_tracker import ScreencopyTracker
+from mir_ci.wayland.virtual_pointer import VirtualPointer
+
+ROBOT_PATH = Path(__file__).parent / "robot"
+
+
+def collect_asset(asset: Path, variant):
+    variants = reversed((variant, *variant.parents[:-1]))
+    return itertools.chain(
+        asset.glob("*"),
+        *((asset / "variants" / v).glob("*") for v in variants),
+    )
+
+
+def collect_assets(platform: str, resources: Collection[str], suite: str, variant: Path = VARIANT):
+    return {
+        p.name: p
+        for p in itertools.chain(
+            collect_asset(ROBOT_PATH / "platforms" / platform, variant),
+            *(collect_asset(ROBOT_PATH / "resources" / resource, variant) for resource in resources),
+            collect_asset(ROBOT_PATH / "suites" / suite, variant),
+        )
+        if p.is_file()
+    }
```

Comment:
Nit: Is this _nearly_ the same as the functions found in `mir-ci/mir_ci/tests/test_osk.py`? Could we abstract that away?

Reply:
We should, yes. We'll also build a separate framework around this so it can be shared outside of mir-ci.

Reply:
Cool cool. So save this for a later time?

Reply:
As this will soon be heavily refactored, I think we can live with the copypasta at the moment.